### DISCUSSION TITLE
you will need to use `defaultValue` for input for `useFieldArray`. especially when you delete, shuffle and swap inputs.

### DISCUSSION THREAD

**Discussion by kayuapi** [react-hook-form/react-hook-form#1756] [General] [answered]
you will need to use `defaultValue` for input for `useFieldArray`. especially when you delete, shuffle and swap inputs.

_Originally posted by @bluebill1049 in https://github.com/react-hook-form/react-hook-form/issues/1529#issuecomment-621081543_

Hi, with regards to this, is there a way so that I can use reset method returned from useForm hook to change the fields returned from useFieldArray hook? From my understanding, useFieldArray takes in control method returned from useFrom hook and parse it to return the fields. However, when I call the reset with useEffect, I am facing issues making the useFieldArray's control parameter to update. My goal is to obtain a new set of fields from useFieldArray when the useForm's defaultValue parameter is updated.

Thank you!

**Comment by bluebill1049** [comment, marked as answer]
please provide a simple codesandbox for your question.

**Reply by kayuapi** [reply]
Hi, thank you for the swift reply!
working version: https://codesandbox.io/s/great-chatterjee-qnc8p?file=/src/App.js
not working version: https://codesandbox.io/s/condescending-cartwright-x8l67?file=/src/App.js

I'm trying to reset the default values and make the resetted default values available to useFieldArray

**Reply by bluebill1049** [reply]
use `reset` api at `useEffect`

**Reply by kayuapi** [reply]
In not working version line 32 to line 34, I used `reset` api at `useEffect`,  but from line 35 to line 38, I cannot make `fields` to have the resetted default values

**Reply by bluebill1049** [reply]
why not working on top of the working version? and add a `useEffect` with `reset`?

**Reply by kayuapi** [reply]
```
  const [receivedItems, setReceivedItems] = useState([]);

  useEffect(() => {
    setTimeout(() => {
      setReceivedItems([{ id: 0, name: "nature5" }]);
    }, 1000);
  }, []);
  useEffect(() => {
    reset({ defaultValues: { items: receivedItems } });
  }, [receivedItems, reset]);
```
Added this to the working version and the `fields` api from `useFieldArray` are not updated to the new default values, the fields just disappeared

**Reply by bluebill1049** [reply]
why is this added? `setReceivedItems([{ id: 0, name: "nature5" }]);`

```
useEffect(() => {
    // pass your data here
    reset({
      items: [] // not defaultValues
    });
  }, [yourItem, reset]);
```

please invest some time with the doc, there are examples there as well to help you understand `useFieldArray` and other APIs.

https://react-hook-form.com/api#reset
https://react-hook-form.com/api#useFieldArray

**Reply by kayuapi** [reply]
It's a mock to simulate a set local component state action upon receiving a promise. 
Thank you! It works!
